### PR TITLE
8344137: Update XML Security for Java to 3.0.5

### PR DESCRIPTION
**Pull request by wangweij**
Update XML Signature to Apache Santuario 3.0.5.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change requires CSR request [JDK-8345411](https://bugs.openjdk.org/browse/JDK-8345411) to be approved

### Issues
 * [JDK-8344137](https://bugs.openjdk.org/browse/JDK-8344137): Update XML Security for Java to 3.0.5 (**Enhancement** - P3)
 * [JDK-8345411](https://bugs.openjdk.org/browse/JDK-8345411): Update XML Security for Java to 3.0.5 (**CSR**)

### Reviewers
 * [Sean Mullan](https://openjdk.org/census#mullan) (@seanjmullan - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/22584/head:pull/22584` \
`$ git checkout pull/22584`

Update a local copy of the PR: \
`$ git checkout pull/22584` \
`$ git pull https://git.openjdk.org/jdk.git pull/22584/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 22584`

View PR using the GUI difftool: \
`$ git pr show -t 22584`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/22584.diff">https://git.openjdk.org/jdk/pull/22584.diff</a>

</details>
<details><summary>Using Webrev</summary>

[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/22584#issuecomment-2521399039)
</details>

### CODE REVIEW

File: src/java.xml.crypto/share/legal/santuario.md [hunk 1]
```diff
@@ -223,5 +223,5 @@ The development of this software was partly funded by the European
 Commission in the <WebSig> project in the ISIS Programme.
 
 This product contains software that is
-copyright (c) 2021, Oracle and/or its affiliates.
+copyright (c) 2021, 2023, Oracle and/or its affiliates.
```
Comment (mcpowers) [inline src/java.xml.crypto/share/legal/santuario.md:226]: Why not 2024?

**Review by mcpowers — COMMENTED** [review]
The only thing I noticed were old Oracle copyright dates in some files with "dom" or "DOM" in their path names. Many contained a single 2005 but at least one had 2005 plus another date that was not 2024.

File: src/java.xml.crypto/share/legal/santuario.md (hunk 1 above)
Comment (wangweij) [reply src/java.xml.crypto/share/legal/santuario.md:226]: This is copy of the [LICENSE](https://github.com/apache/santuario-xml-security-java/blob/main/NOTICE) file from Santuario.

**Comment by wangweij** [comment]
> The only thing I noticed were old Oracle copyright dates in some files with "dom" or "DOM" in their path names. Many contained a single 2005 but at least one had 2005 plus another date that was not 2024.

I don’t update the copyright year when importing a new version. The instances where you see 2005 date back to the Sun Microsystems era. While we updated the company name, the years have remained unchanged. If there is another date, it probably means there was a change made inside OpenJDK.

**Review by seanjmullan — APPROVED** [review]
Looks good. Please write a release note noting the new SignatureMethod constants for ECDSA with SHA-3.

**Comment by wangweij** [comment]
Thanks. I've create the release note at https://bugs.openjdk.org/browse/JDK-8345729.

**Comment by wangweij** [comment]
I've updated the `XMLUtils` helper class for internal testing.

**Review by seanjmullan — APPROVED** [review]

**Pull request closed without merging**